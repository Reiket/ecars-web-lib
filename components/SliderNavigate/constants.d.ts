export declare const NAVIGATE_BUTTON_PLACEMENT: Readonly<{
    LEFT: "left";
    RIGHT: "right";
}>;
export type NavigateButtonPlacement = (typeof NAVIGATE_BUTTON_PLACEMENT)[keyof typeof NAVIGATE_BUTTON_PLACEMENT];
export declare const SLIDER_NAVIGATE_TEST_ID = "sliderNavigateTestId";
export declare const NAVIGATE_BUTTON_TEST_ID = "navigateButtonTestId";
