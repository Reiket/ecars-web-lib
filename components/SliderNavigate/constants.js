const T = Object.freeze({ LEFT: "left", RIGHT: "right" }), e = "sliderNavigateTestId", t = "navigateButtonTestId";
export {
  T as NAVIGATE_BUTTON_PLACEMENT,
  t as NAVIGATE_BUTTON_TEST_ID,
  e as SLIDER_NAVIGATE_TEST_ID
};
//# sourceMappingURL=constants.js.map
