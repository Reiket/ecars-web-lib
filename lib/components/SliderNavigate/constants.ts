export const NAVIGATE_BUTTON_PLACEMENT = Object.freeze({
  LEFT: 'left',
  RIGHT: 'right',
});

export type NavigateButtonPlacement = (typeof NAVIGATE_BUTTON_PLACEMENT)[keyof typeof NAVIGATE_BUTTON_PLACEMENT];

export const SLIDER_NAVIGATE_TEST_ID = 'sliderNavigateTestId';
export const NAVIGATE_BUTTON_TEST_ID = 'navigateButtonTestId';
