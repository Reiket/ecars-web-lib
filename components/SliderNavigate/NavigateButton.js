import { jsx as t } from "react/jsx-runtime";
import { NAVIGATE_BUTTON_TEST_ID as a } from "./constants.js";
import { Icons as r } from "../../services/icons.js";
const e = { left: t(r.ArrowNarrowLeft, {}), right: t(r.ArrowNarrowRight, {}) }, m = ({ placement: o }) => t("button", { "data-testid": a, type: "button", className: "slider-navigate__button", children: e[o] });
export {
  m as NavigateButton
};
//# sourceMappingURL=NavigateButton.js.map
