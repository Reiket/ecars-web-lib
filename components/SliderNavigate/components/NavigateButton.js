import { jsx as t } from "react/jsx-runtime";
import { NAVIGATE_BUTTON_TEST_ID as i } from "../constants.js";
import { Icons as o } from "../../../services/icons.js";
const a = {
  left: /* @__PURE__ */ t(o.ArrowNarrowLeft, {}),
  right: /* @__PURE__ */ t(o.ArrowNarrowRight, {})
}, m = ({ placement: r }) => /* @__PURE__ */ t(
  "button",
  {
    "data-testid": i,
    type: "button",
    className: "slider-navigate__button",
    children: a[r]
  }
);
export {
  m as NavigateButton
};
//# sourceMappingURL=NavigateButton.js.map
