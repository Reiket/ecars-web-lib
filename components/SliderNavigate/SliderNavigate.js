import { jsx as r } from "react/jsx-runtime";
import { NavigateButton as a } from "./NavigateButton.js";
import { SLIDER_NAVIGATE_TEST_ID as i, NAVIGATE_BUTTON_PLACEMENT as e } from "./constants.js";
const N = () => /* @__PURE__ */ r(
  "div",
  {
    "data-testid": i,
    className: "slider-navigate",
    children: Object.values(e).map((t) => /* @__PURE__ */ r(
      a,
      {
        placement: t
      },
      t
    ))
  }
);
export {
  N as SliderNavigate
};
//# sourceMappingURL=SliderNavigate.js.map
