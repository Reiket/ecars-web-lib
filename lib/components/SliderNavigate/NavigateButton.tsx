import type {FC} from 'react';
import React from 'react';
import type {NavigateButtonPlacement} from '@/components/SliderNavigate/constants';
import {NAVIGATE_BUTTON_TEST_ID} from '@/components/SliderNavigate/constants';
import {Icons} from '@/services/icons';

interface Props {
  placement: NavigateButtonPlacement;
}

const icons = {
  left: <Icons.ArrowNarrowLeft />,
  right: <Icons.ArrowNarrowRight />,
};

export const NavigateButton: FC<Props> = ({placement}) => {
  return (
    <button
      data-testid={NAVIGATE_BUTTON_TEST_ID}
      type="button"
      className="slider-navigate__button"
    >
      {icons[placement]}
    </button>
  );
};
