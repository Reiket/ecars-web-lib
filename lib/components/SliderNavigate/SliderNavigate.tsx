import {NavigateButton} from '@/components/SliderNavigate/components/NavigateButton';
import {NAVIGATE_BUTTON_PLACEMENT, SLIDER_NAVIGATE_TEST_ID} from '@/components/SliderNavigate/constants';
import type {FC} from 'react';

export const SliderNavigate: FC = () => (
  <div
    data-testid={SLIDER_NAVIGATE_TEST_ID}
    className="slider-navigate"
  >
    {Object.values(NAVIGATE_BUTTON_PLACEMENT).map((placement) => (
      <NavigateButton
        key={placement}
        placement={placement}
      />
    ))}
  </div>
);
