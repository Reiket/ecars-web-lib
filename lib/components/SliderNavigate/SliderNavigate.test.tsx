import {cleanup, render, screen} from '@testing-library/react';
import {SliderNavigate} from '@/components/SliderNavigate/SliderNavigate';
import {
  NAVIGATE_BUTTON_PLACEMENT,
  NAVIGATE_BUTTON_TEST_ID,
  SLIDER_NAVIGATE_TEST_ID,
} from '@/components/SliderNavigate/constants';
import {NavigateButton} from '@/components/SliderNavigate/NavigateButton';

describe('Slider Navigate component', () => {
  afterEach(cleanup);
  test('renders correctly', () => {
    const {container} = render(<SliderNavigate />);
    const sliderNavigateElement = screen.getByTestId(SLIDER_NAVIGATE_TEST_ID);
    const buttonElements = screen.getAllByTestId(NAVIGATE_BUTTON_TEST_ID);
    expect(buttonElements.length).toBe(Object.values(NAVIGATE_BUTTON_PLACEMENT).length);
    expect(sliderNavigateElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('renders no buttons if NAVIGATE_BUTTON_PLACEMENT is empty', () => {
    jest.spyOn(Object, 'values').mockReturnValue([]);
    const {container} = render(<SliderNavigate />);
    const buttonElements = screen.queryAllByTestId(NAVIGATE_BUTTON_TEST_ID);

    expect(buttonElements.length).toBe(0);
    expect(container).toMatchSnapshot();
  });
});

describe('Navigate Button component', () => {
  const placements = Object.values(NAVIGATE_BUTTON_PLACEMENT);
  placements.forEach((placement) => {
    test(`renders correctly with ${placement} placement`, () => {
      const {container} = render(<NavigateButton placement={placement} />);
      const buttonElement = screen.getByTestId(NAVIGATE_BUTTON_TEST_ID);

      expect(buttonElement).toBeInTheDocument();
      expect(buttonElement).toHaveClass('slider-navigate__button');
      expect(buttonElement.querySelector('svg')).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
});
