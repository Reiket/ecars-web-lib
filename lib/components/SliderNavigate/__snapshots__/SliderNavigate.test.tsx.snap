// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigate Button component renders correctly with left placement 1`] = `
<div>
  <button
    class="slider-navigate__button"
    data-testid="navigateButtonTestId"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
      />
    </svg>
  </button>
</div>
`;

exports[`Navigate Button component renders correctly with right placement 1`] = `
<div>
  <button
    class="slider-navigate__button"
    data-testid="navigateButtonTestId"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
      />
    </svg>
  </button>
</div>
`;

exports[`Slider Navigate component renders correctly 1`] = `
<div>
  <div
    class="slider-navigate"
    data-testid="sliderNavigateTestId"
  >
    <button
      class="slider-navigate__button"
      data-testid="navigateButtonTestId"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
      </svg>
    </button>
    <button
      class="slider-navigate__button"
      data-testid="navigateButtonTestId"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Slider Navigate component renders no buttons if NAVIGATE_BUTTON_PLACEMENT is empty 1`] = `
<div>
  <div
    class="slider-navigate"
    data-testid="sliderNavigateTestId"
  />
</div>
`;

exports[`SliderComponent Navigate component renders correctly 1`] = `
<div>
  <div
    class="slider-navigate"
    data-testid="sliderNavigateTestId"
  >
    <button
      class="slider-navigate__button"
      data-testid="navigateButtonTestId"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
      </svg>
    </button>
    <button
      class="slider-navigate__button"
      data-testid="navigateButtonTestId"
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`SliderComponent Navigate component renders no buttons if NAVIGATE_BUTTON_PLACEMENT is empty 1`] = `
<div>
  <div
    class="slider-navigate"
    data-testid="sliderNavigateTestId"
  />
</div>
`;
